packages/ListBox/src/ListBox.js

```diff
@@ -0,0 +1,164 @@
+  children: PropTypes.node,
+
+  /** Disable the entire ListBox */
+  isDisabled: PropTypes.bool,
+
+  /** Let the user to select multiple options at same time */
+  isMulti: PropTypes.bool,
+
+  /** This options will display the listbox without the Popover */
+  isInline: PropTypes.bool,
+
+  /** Indicates if the popover is visible */
+  isOpen: PropTypes.bool,
+
+  /** Callback returning the current selection on the ListBox */
+  onChange: PropTypes.func,
+
+  /** Defaults label to display when the ListBox has not option selected */
+  placeholder: PropTypes.string,
+
+  /** Indicate which is the height for the options container */
+  height: PropTypes.number,
+};
+
+export const defaultProps = {
+  children: null,
+  Filter: null, // eslint-disable-line
+  Footer: null, // eslint-disable-line
+  height: 200,
+  isDisabled: false,
+  isInline: false,
+  isMulti: false,
+  isOpen: null,
+  onChange: () => {},
+  placeholder: "Select...",
+  Trigger: null, // eslint-disable-line
+};
+
+export function ListBox(props) {
+  const [state, dispatch] = useListBox();
+  const { children, height, placeholder, Trigger: TriggerProps, Footer, Filter } = props;
+
+  const handleEffectHasFooter = effects.handleEffectHasFooter(Footer, dispatch);
+  React.useEffect(handleEffectHasFooter, []);
+
+  const triggerProps = {
+    hasClearButton: true,
+    onClickClear: null,
+    placeholder,
+    onFooterClickAccept: Footer ? Footer.props.onClickAccept : null,
+  };
+
+  const trigger = TriggerProps ? (
+    React.cloneElement(TriggerProps, { ...triggerProps, ...TriggerProps.props })
+  ) : (
+    <Trigger {...triggerProps} />
+  );
+
+  return (
+    <React.Fragment>
+      {trigger}
+      <Content>
+        <Box>
+          {Filter}
+          <List height={height}>
+            <Options>{children}</Options>
+          </List>
+          {Filter ? <NoResults label={Filter.props.noResultsMessage} /> : null}
+          {Footer ? React.cloneElement(Footer, { ref: state.refFooterContainer }) : null}
+        </Box>
+      </Content>
+    </React.Fragment>
+  );
+}
+
+ListBox.propTypes = {
+  ...propTypes,
+  children: PropTypes.node.isRequired,
+  height: PropTypes.number.isRequired,
+  placeholder: PropTypes.string.isRequired,
+};
+
+ListBox.defaultProps = {};
+
+const ListBoxContainer = React.forwardRef((props, ref) => {
+  const [state, dispatch] = useListBox();
+
+  const {
+    children,
+    height,
+    isInline,
+    placeholder,
+    Filter, // eslint-disable-line
+    Trigger, // eslint-disable-line
+    Footer, // eslint-disable-line
+    Popover: PopoverWithProps, // eslint-disable-line
+  } = props;
+
+  const shouldTriggerKeepFocus = PopoverWithProps && PopoverWithProps.props.shouldKeepFocus;
+
+  // IMPERATIVE API
+  const imperativeHandle = handleImperative(state, dispatch);
+  React.useImperativeHandle(ref, imperativeHandle);
+
+  // EFFECTS
+  const handleEffectChildren = effects.handleEffectChildren(props, state, dispatch);
```

Comment:
Just a thought: shouldn't these effects be renamed to start with the `use` prefix?

eg: `useEffectChildren(props, state, dispatch, [props.children])`

Something like: https://reactjs.org/docs/hooks-custom.html#extracting-a-custom-hook

Reply:
Yeah. Custom hooks are the way to go here. All this trickery to pass in state and dispatch would go away.
I think the lesson here is when using a new thing we should read the docs for said thing.
I was pretty surprised to learn that prism didn't know about the eslint plugin for hooks when it's right there in the documentation.

Reply:
indeed, my mistake, I have refactored all this.